### ISSUE TITLE
The size of tensor a (136) must match the size of tensor b (170) at non-singleton dimension 3

### ISSUE THREAD

**Issue by rrroddri** [Haoming02/comfyui-resharpen#6]
Hey, thank you for your custom node, I'm currently using on it with Flux and works very well, but once I try to generate another image with the same reshapen node value , it gives me this error, If I change the value to another one and I queue the prompt, it works without issues, and then I can change to the original value, but as as soon as I try to repeat the workflow, it gives me this error:

`Error occurred when executing SamplerCustomAdvanced:

The size of tensor a (136) must match the size of tensor b (170) at non-singleton dimension 3

  File "G:\ComfyUI\execution.py", line 316, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
  File "G:\ComfyUI\execution.py", line 191, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
  File "G:\ComfyUI\execution.py", line 168, in _map_node_over_list
    process_inputs(input_dict, i)
  File "G:\ComfyUI\execution.py", line 157, in process_inputs
    results.append(getattr(obj, func)(**inputs))
  File "G:\ComfyUI\comfy_extras\nodes_custom_sampler.py", line 612, in sample
    samples = guider.sample(noise.generate_noise(latent), latent_image, sampler, sigmas, denoise_mask=noise_mask, callback=callback, disable_pbar=disable_pbar, seed=noise.seed)
  File "G:\ComfyUI\comfy\samplers.py", line 716, in sample
    output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)
  File "G:\ComfyUI\comfy\samplers.py", line 695, in inner_sample
    samples = sampler.sample(self, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "G:\ComfyUI\comfy\samplers.py", line 600, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "G:\ComfyUI\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "G:\ComfyUI\comfy\k_diffusion\sampling.py", line 655, in sample_dpmpp_2m
    callback({'x': x, 'i': i, 'sigma': sigmas[i], 'sigma_hat': sigmas[i], 'denoised': denoised})
  File "G:\ComfyUI\comfy\samplers.py", line 598, in 
    k_callback = lambda x: callback(x["i"], x["denoised"], x["x"], total_steps)
  File "G:\ComfyUI\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "G:\ComfyUI\custom_nodes\comfyui-resharpen\resharpen.py", line 37, in hijack_callback
    delta = x.detach().clone() - LATENT_CACHE`

I have it connected like this
![resharpen](https://github.com/user-attachments/assets/a0e0373e-1038-47f8-a063-9b03fc2fcb5b)

Thanks for everything

**Comment by rrroddri** [comment]
Thank you for the quickfix, but the issue persists, here's the error output:
`Error occurred when executing SamplerCustomAdvanced:

The size of tensor a (136) must match the size of tensor b (170) at non-singleton dimension 3

  File "G:\ComfyUI\execution.py", line 316, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
  File "G:\ComfyUI\execution.py", line 191, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
  File "G:\ComfyUI\execution.py", line 168, in _map_node_over_list
    process_inputs(input_dict, i)
  File "G:\ComfyUI\execution.py", line 157, in process_inputs
    results.append(getattr(obj, func)(**inputs))
  File "G:\ComfyUI\comfy_extras\nodes_custom_sampler.py", line 612, in sample
    samples = guider.sample(noise.generate_noise(latent), latent_image, sampler, sigmas, denoise_mask=noise_mask, callback=callback, disable_pbar=disable_pbar, seed=noise.seed)
  File "G:\ComfyUI\comfy\samplers.py", line 716, in sample
    output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)
  File "G:\ComfyUI\comfy\samplers.py", line 695, in inner_sample
    samples = sampler.sample(self, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "G:\ComfyUI\comfy\samplers.py", line 600, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "G:\ComfyUI\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "G:\ComfyUI\comfy\k_diffusion\sampling.py", line 655, in sample_dpmpp_2m
    callback({'x': x, 'i': i, 'sigma': sigmas[i], 'sigma_hat': sigmas[i], 'denoised': denoised})
  File "G:\ComfyUI\comfy\samplers.py", line 598, in 
    k_callback = lambda x: callback(x["i"], x["denoised"], x["x"], total_steps)
  File "G:\ComfyUI\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "G:\ComfyUI\custom_nodes\comfyui-resharpen\resharpen.py", line 37, in hijack_callback`

**Comment by Haoming02** [comment]
The original issue is caused by the different resolutions between generations, where the previous cache wasn't cleared, so the size mismatch.

But after the fix, I cannot seem to repro it now...?

**Comment by Haoming02** [comment]
Whether I set the same/different `width` and `height`, or the same/different `details` values, I can't trigger the error.

**Comment by rrroddri** [comment]
I'm sorry, I updated the repo but did not restart ComfyUI, it's working now without issues. Thank you!

**Comment by rrroddri** [comment]
Fixed by repo author.
